kernel/src/mem.rs

```diff
@@ -68,8 +68,8 @@ impl<L, T> AppSlice<L, T> {
         self.len
     }
 
-    pub unsafe fn ptr(&self) -> *const T {
-        self.ptr.ptr.get() as *const T
+    pub fn ptr(&self) -> *const T {
```

Comment:
Getting a pointer to an AppSlice is actually a safe operation because dereferencing it is unsafe, right? I want the value of the pointer to provide to the userland application in order to inform it which buffer its data is in.